src/main/java/org/gradle/profiler/studio/plugin/StudioPluginInstaller.java

```diff
@@ -33,9 +34,8 @@ public void installPlugin(LaunchConfiguration configuration) {
     private static void installPluginToDirectory(Path directory, List<Path> jars) {
         try {
             for (Path jar : jars) {
-                String jarName = jar.getFileName().toString().endsWith(".jar")
-                    ? jar.getFileName().toString()
-                    : jar.getFileName() + ".jar";
+                String jarName = jar.getFileName().toString();
+                checkArgument(jarName.endsWith(".jar"), "Expected jar file: %s to end with .jar", jar);
```

Comment:
Do we really need to check this? Seems like defensive programming.

Reply:
I would leave it since Android Studio doesn't detect the plugins if it doesn't end with .jar. And it's really annoying to debug this. I was there, debugging this once :)

Reply:
Yes, but weren't you there because of that bug causing the file to have no `.jar` extension? Now that it's fixed it's unlikely anyone will have a similar problem. The parameter is called `jar` and everything, it's not like it's unexpected that it has to be called `something.jar` :)

Reply:
Makes sense, removed